xastropy/xguis/igmguesses.py

```diff
@@ -173,17 +206,17 @@ def delete_component(self, component):
         del component
         # Update
         self.velplot_widg.update_model()
-        self.velplot_widg.on_draw(fig_clear=True)
+        self.velplot_widg.on_draw(fig_clear=True,plot_residuals=self.plot_residuals)
```

Comment:
Once self.plot_residuals is defined, we don't need to pass it to on_draw.

That is, you can just have on_draw query whether it is true and false
and act accordingly, e.g.:

if self.plot_residuals:
    do_something()

I suggest you just do that instead of passing it around.
You just need to make sure to initialize to False (or True)
when creating the Class.